src/transformers/benchmark/cache_benchmark.py

```diff
@@ -0,0 +1,180 @@
+
+    def target(self, batch_size, max_cache_length, seq_length, cache_type, mode):
+        self.model.generation_config.max_length = max_cache_length
+        input_ids = (
+            self.tokenizer([self.input_text] * batch_size, return_tensors="pt")
+            .to(self.device)
+            .input_ids[:batch_size, :seq_length]
+        )
+
+        self.model._reset_cache()
+
+        if cache_type == "static":
+            self.model._setup_cache(StaticCache, batch_size, max_cache_len=max_cache_length)
+
+        generated_ids = torch.zeros(batch_size, self.num_iter)
+
+        model = self.model
+        past_key_values = None
+
+        set_seed(0)
+
+        if cache_type == "dynamic":
+            past_key_values = DynamicCache()
+
+        if mode == "compiled":
+            torch.compiler.reset()
+            with torch.no_grad():
+                model = torch.compile(self.model, mode="reduce-overhead", fullgraph=True)
+
+        with torch.no_grad():
+            outputs = model(input_ids, past_key_values=past_key_values, return_dict=False, use_cache=True)
+            input_ids = sample(outputs[0], temperature=0.6, top_k=5)[0]
+            generated_ids[:, 0] = input_ids[:, 0]
```

Comment:
I think we need to make sure we are passing the cache positions as well, otherwise kernels could go faster if we are not writing to the correct positions